pkg/console/controllers/clioidcclientstatus/controller.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+	"github.com/openshift/console-operator/pkg/console/status"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	authnsub "github.com/openshift/console-operator/pkg/console/subresource/authentication"
+	crdsub "github.com/openshift/console-operator/pkg/console/subresource/crd"
+)
+
+// cliOIDCClientStatusController:
+//
+//	writes:
+//	- authentication.config.openshift.io/cluster .status.oidcClients:
+//		- componentName=cli
+//		- componentNamespace=openshift-console
+//		- currentOIDCClients
+//		- conditions:
+//			- Available
+//			- Progressing
+//			- Degraded
+//	- consoles.operator.openshift.io/cluster .status.conditions:
+//		- type=CLIOIDCClientStatusProgressing
+//		- type=CLIOIDCClientStatusDegraded
+//		- type=CLIAuthStatusHandlerProgressing
+//		- type=CLIAuthStatusHandlerDegraded
+type cliOIDCClientStatusController struct {
+	authnLister       configv1listers.AuthenticationLister
+	authStatusHandler *status.AuthStatusHandler
+	crdLister         apiexensionsv1listers.CustomResourceDefinitionLister
+	operatorClient    v1helpers.OperatorClient
+	statusHandler     status.StatusHandler
+}
+
+func NewCLIOIDCClientStatusController(
+	operatorClient v1helpers.OperatorClient,
+	authnInformer configv1informers.AuthenticationInformer,
+	authenticationClient configv1client.AuthenticationInterface,
+	consoleOperatorInformer operatorv1informers.ConsoleInformer,
+	crdInformer apiexensionsv1informers.CustomResourceDefinitionInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &cliOIDCClientStatusController{
+		authnLister:       authnInformer.Lister(),
+		authStatusHandler: status.NewAuthStatusHandler(authenticationClient, api.CLIOIDCClientComponentName, api.TargetNamespace, api.OpenShiftConsoleOperator),
+		crdLister:         crdInformer.Lister(),
+		operatorClient:    operatorClient,
+	}
+	return factory.New().
+		WithSync(c.sync).
+		ResyncEvery(wait.Jitter(time.Minute, 1.0)).
+		WithFilteredEventsInformers(
+			factory.NamesFilter("authentications.config.openshift.io"),
+			crdInformer.Informer(),
+		).
+		WithInformers(
+			authnInformer.Informer(),
+			consoleOperatorInformer.Informer(),
+		).
+		ToController("CLIOIDCClientStatusController", recorder.WithComponentSuffix("cli-oidc-client-status-controller"))
+}
+
+func (c *cliOIDCClientStatusController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	c.statusHandler = status.NewStatusHandler(c.operatorClient)
+	return util.HandleManagementState(ctx, c)
+}
+
+func (c *cliOIDCClientStatusController) GetOperatorSpec() (*operatorv1.OperatorSpec, error) {
+	spec, _, _, err := c.operatorClient.GetOperatorState()
+	return spec, err
+}
```

Comment:
I'd suggest dropping the `GetOperatorSpec()` from the interface and always use the `operatorClient` instead to remove the need for unnecessary code duplication.